ledger/ledger-api-test-tool/src/test/suite/scala/com/daml/ledger/api/testtool/NamesSpec.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool
+
+import com.daml.ledger.api.testtool.NamesSpec._
+import com.daml.ledger.api.testtool.tests.Tests
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class NamesSpec extends AnyWordSpec with Matchers {
+  "test suite names" should {
+    "only contain letters" in {
+      all(allTestSuiteNames) should fullyMatch regex """[A-Za-z]+""".r
+    }
+
+    "not be a prefix of any other name, so that each suite can be included independently" in {
+      allTestSuiteNames.foreach { name =>
+        all(allTestSuiteNames.toSet - name) should not startWith name
+      }
+    }
+  }
+
+  "test identifiers" should {
+    "only contain letters and numbers, and start with a letter" in {
+      all(allTestIdentifiers) should fullyMatch regex """[A-Za-z][A-Za-z0-9]*""".r
+    }
+
+    "not be a prefix of any other name, so that each test can be included independently" in {
+      allTestIdentifiers.foreach { testIdentifier =>
+        all(allTestIdentifiers.toSet - testIdentifier) should not startWith testIdentifier
+      }
+    }
+  }
+
+  "full test names" should {
+    "be unique" in {
+      allTestNames.foreach { name =>
+        allTestNames.filter(_ == name) should have size 1
+      }
```

Comment:
Wouldn't it be simpler to convert `allTestNames` to a set and test for equal length?

Reply:
I tried that but the error message was useless. "expected <some giant list> to have size 999 but was 972", or something. And an equality check resulted in "<some giant list> does not equal <some other giant list>".

This yields an error message that you can actually use to find the problem.

Reply:
Ah yes, makes sense. Not sure if it's worth collecting all the duplicate tests though before failing, as I wouldn't expect to get a lot of duplicates.

Reply:
we already have a diffing lib in the legder API test tool, you could use that if you want but I’m also perfectly happy with this solution.

Reply:
Rule number one of writing tests: make sure it fails.

Rule number two: make sure the failure tells you something useful.